R/register_handlers.R

```diff
@@ -0,0 +1,107 @@
+#' Register handlers for logging messages, warnings and errors
+#'
+#' @param namespace (`character(1)`) the logger namespace
+#' @param package (`character(1)`) the package name
+#'
+#' @return `NULL` invisibly. Called for its side effects.
+#'
+#' @details This function registers global handlers for messages, warnings and errors.
+#' The handlers will investigate the call stack and if it contains a function
+#' from the package, the message, warning or error will be logged into the respective
+#' namespace.
+#' The handlers are registered only once per package and type. Consecutive calls will be ignored.
+#'
+#' @note Registering handlers is forbidden within `tryCatch()` or `withCallingHandlers()`, e.g.
+#' `try(globalCallingHandlers(NULL))` will throw an error.
+#' Because of this, handlers are registered conditionally - only if it is possible.
+#'
+#' @seealso [globalCallingHandlers()]
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' register_handlers("teal.logger")
+#' # see the outcome
+#' globalCallingHandlers()
+#' }
+register_handlers <- function(namespace, package = namespace) {
+  if (if_register_handler_possible()) {
```

Comment:
rename to `is_register_handle_possible` or `register_handler_possible` so you dont have `if (if_` : P